kong/db/schema/entities/upstreams.lua

```diff
@@ -212,6 +213,12 @@ local r =  {
 
     -- different headers
     { distinct = { "hash_on_header", "hash_fallback_header" }, },
+
+    -- if algorithm is "least" then hash_on must be "none"
+    { conditional = {
+      if_field = "algorithm", if_match = { match = "^least$" },
+      then_field = "hash_on", then_match = { one_of = { "none" }, },
```

Comment:
I suggest using `eq = "none"` instead of `one_of = { "none" }` here

Reply:
done